src/dht/dht_server.cc

```diff
@@ -597,7 +597,8 @@ void
 DhtServer::create_error(const DhtMessage& req, const rak::socket_address* sa, int num, const char* msg) {
   DhtMessage error;
 
-  if ((req[key_t].is_raw_bencode() || req[key_t].is_raw_string()) && req[key_t].as_raw_string().size() < 20)
+  if ((req[key_t].is_raw_bencode() && req[key_t].as_raw_bencode().size() < 20) ||
+		  (req[key_t].is_raw_string() && req[key_t].as_raw_string().size() < 20))
```

Comment:
If the key is 20 bytes long string, then the bencode version of it will be '20:<20*bytes>" which is 23 bytes.

However the key can be more than 20 char long strings, so perhaps we should use e.g. 64 and 67 as the max lengths.